src/goto-symex/goto_symex.h

```diff
@@ -711,6 +711,12 @@ class goto_symext
    */
   void replace_nondet(expr2tc &expr);
 
+  /**
+   * Process a block adding the width of each symbol into the stack length
+   * @param expr Expr to search for symbols.
+   */
+  void process_stack_size(expr2tc &expr);
+
```

Comment:
I think this should be part of statet::framet.

Reply:
I put in `goto_symext`because the claim will need to be stack size of the current execution. Currently, each frame only contains the size of its function, the correct claim will contain all frames so far.

Reply:
I don't get it: the new option `stack-limit` defines a global stack limit, i.e., the combined stacks should be smaller than this number?

If that's the case, you can move this to statet::framet and add a new argument: 
```
void process_stack_size(expr2tc &expr, goto_symex_statet *cur_state);
```
and in there you can get the stack limit and call `cur_state->claim`.

Reply:
I have moved `process_stack_size` to `goto_symex_statet::framet`, as suggested by @mikhailramalho. However, I have used a different strategy to deal with the `claim` and `stack_limit`. In particular, I added to `symex_decl` the following implementation:

```
  if(stack_limit != 0)
  {
    claim(
      (cur_state->top().process_stack_size(code2, stack_limit)),
      "Stack limit was violated");
  }
```